.evergreen/config_generator/components/loadbalanced.py

```diff
@@ -0,0 +1,106 @@
+from shrub.v3.evg_build_variant import BuildVariant
+from shrub.v3.evg_command import EvgCommandType, FunctionCall, expansions_update
+from shrub.v3.evg_task import EvgTaskRef, EvgTaskDependency
+
+from config_generator.components.funcs.bootstrap_mongo_orchestration import BootstrapMongoOrchestration
+from config_generator.components.funcs.fetch_det import FetchDET
+from config_generator.components.funcs.upload_build import UploadBuild
+from config_generator.components.funcs.run_simple_http_server import RunSimpleHTTPServer
+from config_generator.components.funcs.fetch_build import FetchBuild
+from config_generator.components.funcs.run_tests import RunTests
+from config_generator.etc.utils import Task, bash_exec
+
+
+def functions():
+    return {
+        'start load balancer': [
+            bash_exec(
+                command_type=EvgCommandType.SETUP,
+                script='''\
+                    export DRIVERS_TOOLS=./drivers-evergreen-tools
+                    export MONGODB_URI="${MONGODB_URI}"
+                    $DRIVERS_TOOLS/.evergreen/run-load-balancer.sh start
+                ''',
+            ),
+            expansions_update(
+                command_type=EvgCommandType.SETUP,
+                file='lb-expansion.yml',
+            )
+        ]
+    }
+
+
+def make_test_task(auth: bool, ssl: bool, server_version: str):
+    auth_str = "auth" if auth else "noauth"
+    ssl_str = "ssl" if ssl else "nossl"
+    return Task(
+        name=f"loadbalanced-test-{auth_str}-{ssl_str}-{server_version}",
+        depends_on=[EvgTaskDependency(name="loadbalanced-compile")],
+        # Use `rhel8.7` distro. `rhel8.7` distro includes necessary dependency: `haproxy`.
+        run_on="rhel8.7-small",
```

Comment:
Consider including `{distro_str}` in the task name (using `make_distro_str()`) for consistency with other new EVG config generator scripts, even if all the (current) tasks are on the same/single distro.

Also recommend specifying the `run_on` distro using `find_small_distro()` from `config_generator.etc.distros` rather than hard-coding the distro string.